jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -40,7 +41,7 @@
   private final Path sourceFile;
   private final AbsoluteUnixPath extractionPath;
   private final FilePermissions permissions;
-  private final long lastModifiedTime;
+  private final FileTimestampProvider modifiedTimeProvider;
```

Comment:
This should remain either `long` or (better) `Instant` — no need to have a provider here.

Reply:
If we only have file timestamp specification on individual files, I don't even think we need this interface anymore. If we use it for `addEntryRecursive`, I think just using `Function` would be more self-documenting.